### ISSUE TITLE
Choco automation doesn't appear to upload new releases

### ISSUE THREAD

**Issue by dvonthenen** [vmware-tanzu/community-edition#3178]
## Bug Report

After an audit of the Choco github action and associated scripts (https://github.com/vmware-tanzu/community-edition/blob/main/hack/choco/update-choco-package.sh), it does appear (or at least it is difficult to find) that the automation is invoking the script (https://github.com/vmware-tanzu/community-edition/blob/main/hack/choco/packpush.ps1) to actually perform the update. Also, not entirely sure that the script (https://github.com/vmware-tanzu/community-edition/blob/main/hack/choco/packpush.ps1) will successfully perform the update. Need a review of the aforementioned script.

## Expected Behavior

The choco package is updated in the choco repo

## Steps to Reproduce the Bug

Please see this failure:
https://github.com/vmware-tanzu/community-edition/actions/runs/1859766349

## Environment Details

* Build version (`tanzu version`): 
* Deployment (Managed/Standalone cluster): 
* Infrastructure Provider (Docker/AWS/Azure/vSphere): 
* Operating System (client):

**Comment by dvonthenen** [comment]
Linking to parent:
https://github.com/vmware-tanzu/community-edition/issues/1663

**Comment by nrb** [comment]
See also #2228; while we're automatically updating some values, there needs to be a job that runs on Windows machines and executes `choco pack/choco push`

**Comment by karuppiah7890** [comment]
Yes, I noticed #2228 and it shouldn't have been closed without the push process 😅

**Comment by aman556** [comment]
Hello everyone, I just saw this issue. 
https://github.com/vmware-tanzu/community-edition/actions/runs/1859766349 
This failure is due to not doing the git Configuration with githubaction which is handled in [#3242](https://github.com/vmware-tanzu/community-edition/pull/3242). Yes the automation has to be edit to do executes `choco pack/choco push`. I will try to update [#3242 ](https://github.com/vmware-tanzu/community-edition/pull/3242).
Thanks

**Comment by aman556** [comment]
Hey @nrb and @dvonthenen,
For [this comment](https://github.com/vmware-tanzu/community-edition/pull/3242#pullrequestreview-892739688) and above this issue I have come up with an approach.
So we have three scripts to take care of choco automation for testing and upgrading.
- Upgrade test - This powershell Script is just testing the current version and then temporarily updating the choco package script with the incoming version and testing it. [Related PRs - [#3242](https://github.com/vmware-tanzu/community-edition/pull/3242) ]
- Update Script - This Script will run to update the [chocolateyinstall.ps1](https://github.com/vmware-tanzu/community-edition/blob/main/hack/choco/tools/chocolateyinstall.ps1) and [tanzu-community-edition.nuspec](https://github.com/vmware-tanzu/community-edition/blob/main/hack/choco/tanzu-community-edition.nuspec). [Related PR's - [#3055](https://github.com/vmware-tanzu/community-edition/pull/3055) [#3210](https://github.com/vmware-tanzu/community-edition/pull/3210)]
- [packpush.ps1](https://github.com/vmware-tanzu/community-edition/blob/main/hack/choco/packpush.ps1) script will run once the above scripts give green signal 

So the above scripts will run in below sequence in one workflow (I will club workflow of [#3242](https://github.com/vmware-tanzu/community-edition/pull/3242) and [#3055](https://github.com/vmware-tanzu/community-edition/pull/3055)) if any script fail then automatically others will stop.
<img width="673" alt="Screenshot 2022-02-25 at 12 44 59 PM" src="https://user-images.githubusercontent.com/56108587/155676914-f9a6b4d1-0bb4-4b00-be9b-18c6d60c8d1a.png">

**Comment by karuppiah7890** [comment]
Like @dvonthenen said

> Also, not entirely sure that the script (https://github.com/vmware-tanzu/community-edition/blob/main/hack/choco/packpush.ps1) will successfully perform the update. Need a review of the aforementioned script.

We do have to review the pack and push script. The script currently uses a local repository to push, notice the `--source`

https://github.com/vmware-tanzu/community-edition/blob/b035580373517291cd4ee88acdcaa3fad9b113d0/hack/choco/packpush.ps1#L18-L20

We need to push the package to the public Chocolatey Community Repository at https://community.chocolatey.org/ which would require some API keys I believe, with access to push the package and we need to inject those in the GitHub Actions Secrets through repo settings and then use them in the command line in our workflow. We can't use the current script as is as that is used for a local development workflow rather than for publishing to the public Chocolatey Community Repository

**Comment by aman556** [comment]
Thanks @karuppiah7890 for mentioning this. I am planning a PR with a comman workflow and working packpush script and will update everyone in this issue during the ongoing work on that PR.

**Comment by nrb** [comment]
`packpush.ps1` was made for local testing, so yes, it does not currently know how to push to the public repos.

As stated before, if the intention is to use it for both it will need to be modified and any flags that change behavior need to be documented.

**Comment by dvonthenen** [comment]
> `packpush.ps1` was made for local testing, so yes, it does not currently know how to push to the public repos.
> 
> As stated before, if the intention is to use it for both it will need to be modified and any flags that change behavior need to be documented.

The original issue description might not be correct since I haven't really been involved with the choco stuff.

If we want to use the same script, great. If we want to have 2 separate scripts for separation of concerns, great. The intention of the issue is to provide automation somewhere in the pipeline to publish to chocolatey.org.

**Comment by joshrosso** [comment]
@karuppiah7890 assigning this one to you for v0.11; feel free to delegate or let us know if it's not realistic for v0.11

**Comment by dvonthenen** [comment]
This should be addressed with this PR https://github.com/vmware-tanzu/community-edition/pull/4346
